### PR TITLE
Dev/simplify api [UP-897]

### PR DESCRIPTION
**Pull request by leroxyl**

### CODE REVIEW

File: ubirch/ubirch_protocol.c [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+/**
+ * Start a new message. Writes the header data to ubirch protocol package.
+ *
+ * @param upp the ubirch protocol context
+ * @param variant protocol variant
+ * @param payload_type the payload data type indicator (0 - binary)
+ * @return -1 if upp is NULL
+ * @return -2 if the protocol version is not supported
+ */
+static int8_t ubirch_protocol_start(ubirch_protocol *upp, ubirch_protocol_variant variant) {
```
Comment (thinkberg) [inline ubirch/ubirch_protocol.c:37]: I am wondering why the structure has changed here from inline code in the header to a static function in the .c file.
Some parts are still in the header, these are not, why?

**Review by thinkberg — CHANGES_REQUESTED** [review]
See comment.

File: ubirch/ubirch_protocol.c (hunk 1 above)
Comment (leroxyl) [reply ubirch/ubirch_protocol.c:37]: I've put the functions that are not part of the public API as static in the .c file for access control

**Comment by thinkberg** [comment]
Awesome then I’m fine :-)
    	
    	

    	-- Matthias L. Jugel | +49 151 2910 9505
    
  




On Fri, Oct 25, 2019 at 2:50 PM +0200, "LaRox11" <notifications@github.com> wrote:












@LaRox11 commented on this pull request.





In ubirch/ubirch_protocol.c:
> + * ```
+ */
+
+#include "ubirch_protocol.h"
+#include "ubirch_protocol_kex.h"
+
+/**
+ * Start a new message. Writes the header data to ubirch protocol package.
+ *
+ * @param upp the ubirch protocol context
+ * @param variant protocol variant
+ * @param payload_type the payload data type indicator (0 - binary)
+ * @return -1 if upp is NULL
+ * @return -2 if the protocol version is not supported
+ */
+static int8_t ubirch_protocol_start(ubirch_protocol *upp, ubirch_protocol_variant variant) {



I've put the functions that are not part of the public API as static in the .c file for access control

**Pull request merged**